### PR TITLE
Services ui rpm

### PR DESCRIPTION
**Pull request by brianhatchl**

### CODE REVIEW

**Comment by brianhatchl** [comment]
make hoot-deps seems to be removing postgis2_92 along with gdal and geos
>sudo yum -y remove gdal-devel gdal geos geos-devel
.
.
.
.
Dependency Removed:
  gdal-libs.x86_64 0:1.9.2-6.rhel6                                              
  libspatialite.x86_64 0:2.4.0-0.6.RC4.el6                                      
  libspatialite-devel.x86_64 0:2.4.0-0.6.RC4.el6                                
  postgis2_92.x86_64 0:2.1.8-1.rhel6                                            
  postgis2_92-client.x86_64 0:2.1.8-1.rhel6                                     

Which is the cause of the past two jenkins failures.

**Comment by brianhatchl** [comment]
I may have misinterpreted the failure of this jenkins test as being javadoc related.  Those may have just been really verbose warnings and that his is the actual culprit
```
/bin/bash: npm: command not found
make[2]: *** [node_modules] Error 127
```
@jasonsurratt trying to figure out why `npm` is being triggered.  Seems to be the same issue with https://github.com/ngageoint/hootenanny/pull/413 now.  I can only find this use in Makefile.old
```
ui-test:
ifeq ($(BUILD_UI),ui)
	cd hoot-ui; npm test #&> /dev/null
endif
```
but it doesn't seem to be getting triggered for me locally (or maybe I'm just running the steps too discretely).

**Comment by jasonsurratt** [comment]
Yeah, npm is also called in docs/Makefile. I'm wondering if there is a race condition between the npm install and the call to npm, but so far I don't see where npm is getting installed. Maybe we just need to add it to `make deps`. I'll make a change in another branch so we can get it merged quickly if that is the case.

**Comment by brianhatchl** [comment]
Ahh, I see that now.  That makes more sense, I couldn't see how ui-test was getting called.  I still don't understand why it's not failing locally for me.

**Comment by brianhatchl** [comment]
Jenkins test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by jasonsurratt** [comment]
Dunno if it helps, but I just ran a clean build on my vbox instance and it builds gdal successfully (yeah!), but I fail with this:

This appears to happen when building the RPM from source. Is `--with-services` being specified when the tarball is created?

```
+ cp LocalConfig.pri.orig LocalConfig.pri
+ command -v ccache
+ make -s -j2
make[2]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
make[3]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
Making all in stub
make[4]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb/stub'
make[5]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
make[5]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
make[5]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
Using default database configuration. You can safely customize: conf/DatabaseConfig.sh
make[6]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb/hoot-services'
../scripts/GenerateBuildInfoFiles.sh: line 15: ../hoot-ui/data/buildInfo.json: No such file or directory
make[6]: *** [build-info] Error 1
make[6]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb/hoot-services'
make[5]: *** [services-build] Error 2
make[5]: *** Waiting for unfinished jobs....
make[5]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb/stub'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb'
error: Bad exit status from /var/tmp/rpm-tmp.PiYak7 (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.PiYak7 (%build)
make[1]: *** [RPMS/x86_64/hootenanny-core-0.2.22_343_gbd03ddb.el6.x86_64.rpm] Error 1
make[1]: Leaving directory `/home/vagrant/hootenanny-rpms/src'
make: *** [hoot-rpms] Error 2
Connection to 127.0.0.1 closed.
make: *** [vagrant-build-rpms] Error 2
```

**Comment by brianhatchl** [comment]
Yes, line https://github.com/ngageoint/hootenanny-rpms/blob/services-ui/BuildArchive.sh#L70

But it seems your error has to do with hoot-ui,
>make[6]: Entering directory `/home/vagrant/hootenanny-rpms/src/BUILD/hootenanny-0.2.22_343_gbd03ddb/hoot-services'
../scripts/GenerateBuildInfoFiles.sh: line 15: **../hoot-ui/data/buildInfo.json**: No such file or directory

**Comment by jasonsurratt** [comment]
Yeah, it is hoot-ui related, but the hoot-ui submodule in my tmp/hootenanny directory was empty so those files never made it into the tar ball. I cleaned out that directory and I'll give it another run.

**Comment by jasonsurratt** [comment]
Jenkins, test this please.

**Comment by jasonsurratt** [comment]
I stopped the above job so that it could be run with a clean-all first. The Jenkins job is now configured to run clean-all with each build. We can turn that off later if need be.

**Comment by jasonsurratt** [comment]
Got past gdal step without error!

**Comment by brianhatchl** [comment]
Woot!  Test passed.  @jasonsurratt can you review and merge?

File: BuildArchive.sh [hunk 1]
```diff
@@ -5,8 +5,11 @@
 set -e
 set -x
 
+export GIT_COMMIT=origin/rpm-services-ui
```
Comment (brianhatchl) [inline BuildArchive.sh]: Will want to delete this line when changes in hoot's rpm-services-ui branch are merged into develop.

**Comment by jasonsurratt** [comment]
Wow!!! Awesome!

**Comment by jasonsurratt** [comment]
Jenkins, test this please.

**Comment by brianhatchl** [comment]
My vsphere settings don't seem to take effect.  Do I need to edit the VagrantfileLocal on patty?  Or do I need to edit the template /Templates/Vagrant/CentOS6.7-x86_64?

**Comment by jasonsurratt** [comment]
I also modified Patty's vagrant file local to include the "many deps". This should speed up provisioning too.

**Comment by brianhatchl** [comment]
I can't say it's looking any faster with cpu=4 and ram=8gb

**Comment by brianhatchl** [comment]
Jenkins, test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by brianhatchl** [comment]
Commit message should read, until #36 is fixed

**Comment by brianhatchl** [comment]
test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by brianhatchl** [comment]
test this please

**Comment by jasonsurratt** [comment]
I know what it is. I'll fix it. Ugh.

**Comment by brianhatchl** [comment]
test this please

**Comment by jasonsurratt** [comment]
Schweet

**Comment by brianhatchl** [comment]
So is this ready to merge at this point?  
I'm creating tickets for the couple tests still excluded during rpm build.  There is some more work to do 
* finalizing rpm remove behavior
* moving rpm install hook scripts back inside rpm binary
* getting jenkins rpm update test job created

**Comment by jasonsurratt** [comment]
Yup. I think we're good.

**Pull request merged**